code/game/objects/items/weapons/tools.dm

```diff
@@ -791,26 +791,52 @@
 		else
 			to_chat(user, SPAN_WARNING("The cell isn't charged!"))
 
+/obj/item/steelwool/fire_act()
+	ignite()
+
+/obj/item/steelwool/afterattack(atom/A, mob/user)
+	if(istype(A, /obj) || istype(A, /turf))
+		file_down(A, user)
+	return
 
 /obj/item/steelwool/proc/ignite(var/L, mob/user)
-	lit = TRUE
-	user.visible_message(SPAN_NOTICE("[user] ignites the steel wool with \the [L]."), SPAN_NOTICE("You ignite the steel wool."), SPAN_NOTICE("You hear a gentle flame crackling."))
-	playsound(get_turf(src), 'sound/items/flare.ogg', 50)
-	desc += " Watch your hands!"
-	icon_state = "burning_wool"
-	set_light(2, 2, LIGHT_COLOR_LAVA)
-	addtimer(CALLBACK(src, .proc/endburn, user), 120 SECONDS, TIMER_UNIQUE)
-
-/obj/item/steelwool/proc/endburn(mob/living/carbon/human/user)
+	if(lit && user)
+		to_chat(user, SPAN_WARNING("The steel wool is already lit!"))
+		return
+	else
+		lit = TRUE
+		if(user)
+			user.visible_message(SPAN_NOTICE("[user] ignites the steel wool with \the [L]."), SPAN_NOTICE("You ignite the steel wool with \the [L]"), SPAN_NOTICE("You hear a gentle flame crackling."))
+		playsound(get_turf(src), 'sound/items/flare.ogg', 50)
+		desc += " Watch your hands!"
+		icon_state = "burning_wool"
+		set_light(2, 2, LIGHT_COLOR_LAVA)
+		addtimer(CALLBACK(src, .proc/endburn), 120 SECONDS, TIMER_UNIQUE)
+
+/obj/item/steelwool/proc/endburn()
 	visible_message(SPAN_NOTICE("The steel wool burns out."))
-	if(!user.gloves)
-		var/UserLoc = get_equip_slot()
-		if(UserLoc == slot_l_hand)
-			user.apply_damage(5, BURN, BP_L_HAND)
-			to_chat(user, SPAN_DANGER("The steel wool burns your left hand!"))
-		else if(UserLoc == slot_r_hand)
-			user.apply_damage(5, BURN, BP_R_HAND)
-			to_chat(user, SPAN_DANGER("The steel wool burns your right hand!"))
+	if(istype(loc, /mob/living))
+		var/mob/living/carbon/human/user = loc
+		if(!user.gloves)
+			var/UserLoc = get_equip_slot()
+			if(UserLoc == slot_l_hand)
+				user.apply_damage(5, BURN, BP_L_HAND)
+				to_chat(user, SPAN_DANGER("The steel wool burns your left hand!"))
+			else if(UserLoc == slot_r_hand)
+				user.apply_damage(5, BURN, BP_R_HAND)
+				to_chat(user, SPAN_DANGER("The steel wool burns your right hand!"))
 	
 	new /obj/effect/decal/cleanable/ash(get_turf(src))
-	qdel(src)
\ No newline at end of file
+	qdel(src)
+
+/obj/item/steelwool/proc/file_down(atom/target, mob/user)
+	if(target.color != initial(target.color))
+		if(lit)
+			to_chat(user, SPAN_WARNING("The steel wool is too hot to do that!"))
+			return
+		else
+			user.visible_message(SPAN_NOTICE("[user] starts filing down \the [target] with [name]."), SPAN_NOTICE("You start filing down \the [target] with [name]."))
+			if(do_after(user, 2 SECONDS))
+				target.color = initial(target.color)
```

Comment:
This has some problems:

1 - It won't actually update the target's icon because you didn't call update_icon on the target after the color change.
2 - Some things have specific color changing procs, e.g. pipes with their `change_color()`

I think you should remove this `file_down()` proc, it's just going to cause problems.

Alternatively make it do cleaning the same as soap does.

Reply:
In regards to the first point - it seemed to work fine when I was testing it, so I'm not sure what the issue is? But the second point is valid so I'll swap it around.

Reply:
So as an update -I think I'm going to remove the proc altogether. I'd find it rather werid if steel wool could clean everything just like soap would, and I don't think there's any truly good solution to paint not being cleanable on objs either.